### PR TITLE
Fix user-agent register timing in Unity SDK

### PR DESCRIPTION
**Pull request by chkuang-g**
Make sure the user-agents are registered before C++ App is created.

Depends on a new utility function introduced in https://github.com/firebase/firebase-cpp-sdk/pull/1217

***

E2E test using Auth iOS/Android testapp
***

### Type of Change
Place an `x` the applicable box:
- [x] Bug fix. Add the issue # below if applicable.
- [ ] New feature. A non-breaking change which adds functionality.
- [ ] Other, such as a build process or documentation change.
***

### CODE REVIEW

File: app/src/swig/app.i [hunk 1]
```diff
@@ -167,6 +168,49 @@ static void LogHeartbeatForDesktop(App* app) {
 #endif  // FIREBASE_PLATFORM_DESKTOP
 }
 
+static void RegisterLibraryHelper(const char* library, const char* version) {
```
Comment (AlmostMatt) [inline app/src/swig/app.i:171]: Can we delete RegisterLibraryInternal and RegisterLibraryHelper?
It looks like there is no current usage of RegisterLibraryInternal.

Alternatively, if RegisterLibraryInternal is still useful, can it be changed to just create a map with a single entry and then call RegisterLibrariesInternal?

That way we would not need to have these two very similar helpers.

File: app/src/swig/app.i [hunk 2]
```diff
@@ -832,6 +878,27 @@ static firebase::AppOptions* AppOptionsLoadFromJsonConfig(const char* config) {
     Firebase.Platform.FirebaseHandler.Create(
         Firebase.Platform.FirebaseAppUtils.Instance);
     FirebaseApp newProxy;
+
+    // If this is the first App, register library information.
+    if (!userAgentRegistered) {
+        userAgentRegistered = true;
+        var userAgentMap = new StringStringMap();
```
Comment (AlmostMatt) [inline app/src/swig/app.i:885]: since 'new' is used here, I think the map should be deleted after the call to RegisterLibrariesInternal

**Review by AlmostMatt — COMMENTED** [review]
This looks pretty good to me.

File: app/src/swig/app.i (hunk 2 above)
Comment (chkuang-g) [reply app/src/swig/app.i:871]: This is C# code block so userAgentMap would be simply GCed.

File: app/src/swig/app.i (hunk 1 above)
Comment (chkuang-g) [reply app/src/swig/app.i]: Good call. Deleted.

**Review by a-maurice — APPROVED** [review]

**Review by jonsimantov — APPROVED** [review]

**Pull request merged**